### PR TITLE
[SCL-146]  Excel species report endpoint

### PR DESCRIPTION
**Pull request by gridcell**

### CODE REVIEW

**Comment by gridcell** [comment]
[Downloadable Excel Report](https://trello.com/c/x9mphQ3O/146-downloadable-excel-report)

File: src/api/reports/views.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            total_habitat_area += habitat_area
+            total_protected_area += lss.get("protected_area")
+
+            table_data.append([
+                num_landscapes,
+                habitat_area,
+                percent_protected_area,
+            ])
+        
+        if total_habitat_area:
```
Comment (saanobhaai) [inline src/api/reports/views.py:80]: Should protect against dividing by 0

File: src/api/reports/stats.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    protected_area = 0
+
+    for rec in data:
+        num_landscapes += 1
+        habitat_area += rec.area
+        biome_areas = rec.biome_areas or []
+        protected_area += sum([ba.get("protected") or 0 for ba in biome_areas])
+
+    try:
+        percent_protected_area = float(protected_area) / float(habitat_area)
```
Comment (saanobhaai) [inline src/api/reports/stats.py:16]: should protect against dividing by 0

File: src/api/reports/stats.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        ),
+        fragment_landscape_stats=calc_fragment_landscape_stats(country, date, species),
+        restoration_landscape_stats=calc_restoration_landscape_stats(
+            country, date, species
+        ),
+        survey_landscape_stats=calc_survey_landscape_stats(country, date, species),
+    )
+
+
+def calc_conservation_landscape_stats(country, date, species):
```
Comment (saanobhaai) [inline src/api/reports/stats.py:41]: Could conceivably dry some of this up, but not important

File: src/api/reports/views.py (hunk 1 above)
Comment (gridcell) [reply src/api/reports/views.py:80]: if `total_habitat_area` is 0 it won't go in the if statement

File: src/api/reports/stats.py (hunk 2 above)
Comment (gridcell) [reply src/api/reports/stats.py:16]: I am via the exceptions I'm capturing

File: src/api/reports/stats.py (hunk 3 above)
Comment (gridcell) [reply src/api/reports/stats.py:41]: I thought about DRYing up more but thought it was easier to read

**Pull request merged**